src/factories/nodes.ts

```diff
@@ -160,12 +168,16 @@ export async function nodesContainerFactory(runId: string) {
   }
 
   function setPositions(): void {
+    if (!nodesLayout) {
+      return
+    }
+
     for (const [nodeId, node] of nodes) {
-      const position = layout.get(nodeId)
+      const position = nodesLayout.positions.get(nodeId)
 
       if (!position) {
         console.warn(`Could not find node in layout: Skipping ${nodeId}`)
-        return
+        continue
```

Comment:
I found a few places we were returning in a for loop rather than continuing. Unrelated to the feature work but didn't make a separate PR.